app/controllers/posts_controller.rb

```diff
@@ -8,8 +8,64 @@ def index
   def show
     @user = User.find(params[:user_id])
     @user_post = @user.posts.find(params[:id])
+    @author_id = current_user.id
     @post_index = @user.posts.order(created_at: :desc).pluck(:id).index(@user_post.id) + 1
     @likes_count = @user_post.likes_counter
     @comments_count = @user_post.comments_counter
+    @comment = Comment.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    if @post.save
+      flash[:success] = 'New post uploaded successfully!'
+      redirect_to user_post_url(current_user, @post)
+    else
+      flash[:error] = 'Post upload failed! Please try again.'
+      redirect_to new_user_post_url(current_user)
+    end
+  end
+
+  def create_comment
+    @user_post = Post.find(params[:id])
+    @comment = @user_post.comments.build(comment_params)
+    @comment.author = current_user
+
```

Comment:
- [ ] Dear @Caren-Koroeny, the action for creating a comment should be implemented inside the `comment_controller`. Remember, the comments controller should have `new` and `create` actions. implementing this action inside your comment_controller makes your app align with the rails standard and makes your code more readable. 

- [ ] Beside the controller, according to the requirement, There should be a form in the views/comments/new.html.erb file which accepts input for creating comments. This makes your comment stand by its own instead of being rendered inside the post. 

- [Optional] It would be wise to add a private method for filtering your params to make your app even more secure. Please refer [here for more ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) about param methods.